Resources/public/js/views/ez-userprofileview.js

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-userprofileview', function (Y) {
+    'use strict';
+    /**
+     * Provides the User Profile View class
+     *
+     * @module ez-userprofileview
+     */
+    Y.namespace('eZ');
+
+    var CLASS_USER_PROFILE_HIDDEN = 'is-user-profile-hidden',
+        SELECTOR_USER_NAME = '.ez-user-profile-name',
+        SELECTOR_USER_ROLE = '.ez-user-profile-role',
+        SELECTOR_USER_IMAGE = '.ez-user-profile-image';
+
+    /**
+     * The User Profile View
+     *
+     * @namespace eZ
+     * @class UserProfileView
+     * @constructor
+     * @extends eZ.TemplateBasedView
+     */
+    Y.eZ.UserProfileView = Y.Base.create('userProfileView', Y.eZ.TemplateBasedView, [], {
+        events: {
+            '.ez-user-profile': {
+                'tap': '_toggleUserMenu'
+            }
+        },
+
+        initializer: function () {
+           this.after('*:userInfoChange', this._setUserInfo, this);
+        },
+
+        /**
+        * Renders the user profile view
+        *
+        * @method render
+        * @return {eZ.UserProfileView} the view itself
+        */
+        render: function () {
+            this.get('container')
+                .setHTML(this.template())
+                .addClass(CLASS_USER_PROFILE_HIDDEN);
+
+            this.fire('getUserInfo');
+            this._renderUserMenu();
+
+            return this;
+        },
+
+        /**
+        * Renders the user menu view
+        *
+        * @method _renderUserMenu
+        */
+        _renderUserMenu: function () {
+            var view = this.get('userMenuView');
+
+            view.addTarget(this);
+            this.get('container').append(view.render().get('container'));
+            view.set('active', true);
+        },
+
+        /**
+         * Sets name, role and image for the user
+         *
+         * @method _setUserInfo
+         * @protected
+         */
+        _setUserInfo: function () {
+            var user = this.get('userInfo'),
+                container = this.get('container'),
+                fields = user.userData.Version.Fields.field,
+                imageUrl,
+                imageAltText;
+
+                fields.forEach(function (field) {
+                    if (field.fieldDefinitionIdentifier === 'image' && field.fieldValue) {
+                        imageUrl = field.fieldValue.uri;
+                        imageAltText = field.fieldValue.fileName;
```

Comment:
the original filename is most likely a bad alternative text, this should rather be something like *User's avatar image*